src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/TripleDesImplementation.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography;
+
+namespace Internal.Cryptography
+{
+    internal sealed partial class TripleDesImplementation : TripleDES
+    {
+        private const int BitsPerByte = 8;
+        private static readonly RandomNumberGenerator s_rng = RandomNumberGenerator.Create();
+
+        public override KeySizes[] LegalKeySizes
+        {
+            get
+            {
+                // CNG does not support 128-bit keys.
+                // Only support 192-bit keys on all platforms for simplicity.
+                return new KeySizes[] { new KeySizes(minSize: 3 * 64, maxSize: 3 * 64, skipSize: 0) };
+            }
+        }
+
+        public override ICryptoTransform CreateDecryptor()
+        {
+            return CreateTransform(this.Key, this.IV, encrypting: false);
+        }
+
+        public override ICryptoTransform CreateDecryptor(byte[] rgbKey, byte[] rgbIV)
+        {
+            return CreateTransform(rgbKey, rgbIV.CloneByteArray(), encrypting: false);
+        }
+
+        public override ICryptoTransform CreateEncryptor()
+        {
+            return CreateTransform(this.Key, this.IV, encrypting: true);
+        }
+
+        public override ICryptoTransform CreateEncryptor(byte[] rgbKey, byte[] rgbIV)
+        {
+            return CreateTransform(rgbKey, rgbIV.CloneByteArray(), encrypting: true);
+        }
+
+        public override void GenerateIV()
+        {
+            byte[] iv = new byte[BlockSize / BitsPerByte];
+            s_rng.GetBytes(iv);
+            IV = iv;
+        }
+
+        public sealed override void GenerateKey()
+        {
+            byte[] key = new byte[KeySize / BitsPerByte];
+            s_rng.GetBytes(key);
+            Key = key;
+        }
+
+        private ICryptoTransform CreateTransform(byte[] rgbKey, byte[] rgbIV, bool encrypting)
+        {
+            // note: rgbIV is guaranteed to be cloned before this method, so no need to clone it again
```

Comment:
FYI - from my other PR #4932, I've decided to take the approach of don't clone unnecessarily.  Doing that here now, and will fix the existing code in #4932 separately.